src/main/java/ai/elimu/util/AudioMetadataExtractionHelper.java

```diff
@@ -1,47 +1,44 @@
 package ai.elimu.util;
 
+import java.io.File;
 import java.io.IOException;
-import java.io.InputStream;
-import java.util.Map;
-import javax.sound.sampled.AudioFileFormat;
-import javax.sound.sampled.AudioSystem;
-import javax.sound.sampled.UnsupportedAudioFileException;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.tritonus.share.sampled.file.TAudioFileFormat;
+
+
+import org.jaudiotagger.audio.AudioFile;
+import org.jaudiotagger.audio.AudioFileIO;
+import org.jaudiotagger.audio.AudioHeader;
+import org.jaudiotagger.audio.exceptions.CannotReadException;
+import org.jaudiotagger.audio.exceptions.InvalidAudioFrameException;
+import org.jaudiotagger.audio.exceptions.ReadOnlyFileException;
+import org.jaudiotagger.tag.TagException;
 
 /**
  * Utility class for extraction information from audio files.
  */
 public class AudioMetadataExtractionHelper {
-    
+
     private static Logger logger = LogManager.getLogger();
-    
+
     /**
      * Extracts the duration of an audio file.
      */
-    public static Long getDurationInMilliseconds(InputStream inputStream) {
+    public static Long getDurationInMilliseconds(File target) {
         logger.info("getDurationInMilliseconds");
-        
+
         Long duration = null;
         try {
-            AudioFileFormat audioFileFormat = AudioSystem.getAudioFileFormat(inputStream);
-            logger.info("audioFileFormat: " + audioFileFormat);
-            logger.info("audioFileFormat.getType(): " + audioFileFormat.getType());
-            if (!(audioFileFormat instanceof TAudioFileFormat)) {
-                throw new UnsupportedAudioFileException();
-            } else {
-                Map<String, Object> properties = ((TAudioFileFormat) audioFileFormat).properties();
-                if (properties.containsKey("duration")) {
-                    Long durationMicroseconds = (Long) properties.get("duration");
-                    Long durationMilliseconds = durationMicroseconds / 1000;
-                    duration = durationMilliseconds;
-                }
-            }
-        } catch (UnsupportedAudioFileException | IOException ex) {
+            AudioFile audioFile = AudioFileIO.read(target);
+            AudioHeader audioHeader = audioFile.getAudioHeader();
+            // Convert Duration to MilliSeconds.
+            duration = (long)audioHeader.getTrackLength() * 1000;
+        } catch (IOException | CannotReadException | TagException |
+                ReadOnlyFileException | InvalidAudioFrameException ex) {
+            ex.printStackTrace();
```

Comment:
```suggestion
```

Please use Log4j instead of `ex.printStackTrace();`

This is because we are directing log output to log files in the `TEST`/`PROD` environment: https://github.com/elimu-ai/webapp/tree/master/src/main/resources